modules/motions/hooks/usePeriodLimitsInfo.ts

```diff
@@ -0,0 +1,192 @@
+import moment from 'moment'
+import { LimitCheckerAbi, EasyTrackAbi } from 'generated'
+import { createContractHelpers } from 'modules/blockChain/utils/createContractHelpers'
+import {
+  ContractLegoDAIRegistry,
+  ContractEasyTrack,
+  ContractAllowedRecipientRegistry,
+  ContractLegoLDORegistry,
+  ContractRccDAIRegistry,
+  ContractAtcDAIRegistry,
+  ContractGasFunderETHRegistry,
+  ContractPmlDAIRegistry,
+} from 'modules/blockChain/contracts'
+import { useWeb3 } from 'modules/blockChain/hooks/useWeb3'
+import { useSWR, SWRResponse } from 'modules/network/hooks/useSwr'
+import {
+  getLimits,
+  LimitsType,
+  getPeriodData,
+  PeriodDataType,
+} from 'modules/motions/utils'
+import { MotionType } from 'modules/motions/types'
+import { EvmUnrecognized } from 'modules/motions/evmAddresses'
+
+type ContractLimitsMethods = {
+  getLimitParameters: LimitCheckerAbi['getLimitParameters']
+  getPeriodState: LimitCheckerAbi['getPeriodState']
+}
+
+type UsePeriodLimitInfoProps<T> = {
+  address: string
+  contract: T
+  swrKey: string
+  isPending?: boolean
+}
+
+export type usePeriodLimitsInfoResultData = {
+  limits: LimitsType
+  periodData: PeriodDataType
+  motionDuration: number
+  isEndInNextPeriod: boolean
+}
+
+type UsePeriodLimitInfo = <T extends ContractLimitsMethods>(
+  data: UsePeriodLimitInfoProps<T>,
+) => SWRResponse<{
+  limits: LimitsType
+  periodData: PeriodDataType
+  motionDuration: number
+  isEndInNextPeriod: boolean
+}>
+
+const getNewPeriod = ({
+  periodLimit,
+  periodDurationMonths,
+  newPeriodStartTime,
+}: {
+  periodLimit: string
+  periodDurationMonths: number
+  newPeriodStartTime: moment.Moment
+}) => {
+  return {
+    alreadySpentAmount: '0',
+    periodStartTimestamp: newPeriodStartTime.unix(),
+    periodEndTimestamp: newPeriodStartTime
+      .add(periodDurationMonths, 'M')
+      .startOf('month')
+      .unix(),
+    spendableBalanceInPeriod: periodLimit,
+  }
+}
+
+const getPeriodLimitsInfo = async <T extends ContractLimitsMethods>(
+  easyTrack: EasyTrackAbi,
+  contract: T,
+  isPending?: boolean,
+) => {
+  const motionDuration = await easyTrack.motionDuration()
+  const limits = await getLimits(contract)
+  let periodData = await getPeriodData(contract)
```

Comment:
Combining this three await functions in `Promise.all` can help whole function to perform faster, also rpc batching will combine them into one request.